src/linear_algebra/finsupp.lean

```diff
@@ -418,7 +418,7 @@ end total
 protected def dom_lcongr
   {α₁ : Type*} {α₂ : Type*} (e : α₁ ≃ α₂) :
   (α₁ →₀ M) ≃ₗ[R] (α₂ →₀ M) :=
-(finsupp.dom_congr e).to_linear_equiv
+(finsupp.dom_congr e).to_equiv.to_linear_equiv
```

Comment:
Could you please add `add_equiv.to_linear_equiv` that asks only for `map_smul`? Then `(finsupp.dom_congr e).to_linear_equiv (lmap_domain M R e).map_smul` should work here.

Reply:
Is there a reason that `linear_equiv` does not extend `add_equiv`?

Reply:
Either way, out of scope for this PR.

Reply:
When it was written, `linear_map` and `add_monoid_hom` used very different API.